src/Translator/assets/dist/translator_controller.js

```diff
@@ -0,0 +1,229 @@
+import { IntlMessageFormat } from 'intl-messageformat';
+
+function formatIntl(id, parameters = {}, domain, locale) {
+    if (id === '') {
+        return '';
+    }
+    const intlMessage = new IntlMessageFormat(id, [locale.replace('_', '-')], undefined, { ignoreTag: true });
+    Object.entries(parameters).forEach(([key, value]) => {
+        if (key.includes('%') || key.includes('{')) {
+            delete parameters[key];
+            parameters[key.replace(/[%{} ]/g, '').trim()] = value;
+        }
+    });
+    return intlMessage.format(parameters);
+}
+
+function strtr(string, replacePairs) {
+    const regex = [];
+    const map = {};
+    Object.entries(replacePairs).forEach(([from, to]) => {
+        regex.push(from.replace(/([-[\]{}()*+?.\\^$|#,])/g, '\\$1'));
+        map[from] = to;
+    });
+    if (regex.length === 0) {
+        return string;
+    }
+    return string.replace(new RegExp(regex.join('|'), 'g'), (matched) => map[matched].toString());
+}
```

Comment:
You're rebuilding a map that is the exact same as your parameter, here is a suggestion to simplify:

```ts
function sample(string, replacePairs) {
  const regex = [];
  const map = {};
  Object.entries(replacePairs).forEach(([from, to]) => {
    regex.push(from.replace(/([-[\]{}()*+?.\\^$|#,])/g, '\\$1'));
    map[from] = to;
  });
  if (regex.length === 0) {
    return string;
  }
  return string.replace(new RegExp(regex.join('|'), 'g'), (matched) => map[matched].toString());
}

function rework(string, replacePairs) {
  const regex = Object.entries(replacePairs).map(([from, _to]) => {
    return from.replace(/([-[\]{}()*+?.\\^$|#,])/g, '\\$1');
  });
  if (regex.length === 0) {
    return string;
  }
  return string.replace(new RegExp(regex.join('|'), 'g'), (matched) => replacePairs[matched].toString());
}

const samples = [
        // https://github.com/php/php-src/blob/master/ext/standard/tests/strings/strtr_basic.phpt
        ['TEST STrTr', 'test strtr', {'t': 'T', 'e': 'E', 'st': 'ST'}],
        ['TEST STrTr', 'test strtr', {'t': 'T', 'e': 'E', 'st': 'ST'}],

        // https://github.com/php/php-src/blob/master/ext/standard/tests/strings/strtr_variation1.phpt
        ['a23', '123', {'1': 'a', 'a': '1', '2b3c': 'b2c3', 'b2c3': '3c2b'}],
        ['1bc', 'abc', {'1': 'a', 'a': '1', '2b3c': 'b2c3', 'b2c3': '3c2b'}],
        ['a1b2c3', '1a2b3c', {'1': 'a', 'a': '1', '2b3c': 'b2c3', 'b2c3': '3c2b'}],
        [`
a23
1bc
a1b2c3`, `
123
abc
1a2b3c`, {1: 'a', 'a': '1', '2b3c': 'b2c3', 'b2c3': '3c2b'}],

        // https://github.com/php/php-src/blob/master/ext/standard/tests/strings/strtr_variation2.phpt
        ['$', '%', {'$': '%', '%': '$', '#*&@()': '()@&*#'}],
        ['#%*', '#$*', {'$': '%', '%': '$', '#*&@()': '()@&*#'}],
        ['text & @()', 'text & @()', {'$': '%', '%': '$', '#*&@()': '()@&*#'}],
        [`
$
#%*&
text & @()`, `
%
#$*&
text & @()`, {'$': '%', '%': '$', '#*&@()': '()@&*#'}],
    ]


console.log('rework', rework(samples[7][1], samples[7][2]));

console.log('sample', sample(samples[7][1], samples[7][2]));
```

A JSFiddle to test quickly https://jsfiddle.net/kpfthv6u/8/

Reply:
Fixed, thanks